cpp/include/cudf/io/experimental/hybrid_scan.hpp

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file hybrid_scan.hpp
+ */
+
+#pragma once
+
+#include <cudf/io/parquet.hpp>
+#include <cudf/io/parquet_schema.hpp>
+#include <cudf/io/text/byte_range_info.hpp>
+#include <cudf/io/types.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/export.hpp>
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+namespace CUDF_EXPORT cudf {
+namespace io::parquet::experimental {
+
+namespace detail {
+
+/**
+ * @brief Internal experimental Parquet reader optimized for highly selective filters (Hybrid Scan
+ * operation).
+ */
+class impl;
+
+}  // namespace detail
+
```

Comment:
Probably should add these to a doxygen group as well and remove/move the `@file`
```
/**
 * @addtogroup io_readers
 * @{
 * @file
 */
```

Reply:
Just the public header declaration and not the detail namespace or src files.

Reply:
Locally built `docs` and verified that the stuff from `parquet_schema.hpp` and `hybrid_scan.hpp` is being properly added in there